src/Renderers/SpanNodeRenderer.php

```diff
@@ -53,18 +52,25 @@ public function link(?string $url, string $title, array $attributes = []): strin
         }
 
         if (!$this->isSafeUrl($url)) {
-            $attributes['rel'] = 'external noopener noreferrer';
-            $attributes['target'] = '_blank';
+            $attributes = $this->addAttributesForUnsafeUrl($attributes);
         }
 
-        return $this->templateRenderer->render(
-            'link.html.twig',
-            [
-                'url' => $this->environment->generateUrl($url),
-                'title' => $title,
-                'attributes' => $attributes,
-            ]
-        );
+        return $this->decoratedSpanNodeRenderer->link($url, $title, $attributes);
+    }
+
+    public function reference(ResolvedReference $reference, array $value): string
+    {
+        if (!$this->isSafeUrl($reference->getUrl())) {
+            $reference = new ResolvedReference(
+                $reference->getFile(),
+                $reference->getTitle(),
+                $reference->getUrl(),
+                $reference->getTitles(),
+                $this->addAttributesForUnsafeUrl($reference->getAttributes())
+            );
+        }
+
+        return $this->decoratedSpanNodeRenderer->reference($reference, $value);
```

Comment:
this part was a bit tricky because `BaseSpanNodeRenderer::reference()` internally calls `BaseSpanNodeRenderer::link()` thus I had to update the attributes here inside the reference